search-impl/src/main/java/com/example/auction/search/impl/SearchServiceImpl.java

```diff
@@ -0,0 +1,131 @@
+package com.example.auction.search.impl;
+
+import akka.Done;
+import akka.stream.javadsl.Flow;
+import com.example.auction.bidding.api.BidEvent;
+import com.example.auction.bidding.api.BidEvent.BidPlaced;
+import com.example.auction.bidding.api.BidEvent.BiddingFinished;
+import com.example.auction.bidding.api.BiddingService;
+import com.example.auction.item.api.ItemEvent;
+import com.example.auction.item.api.ItemEvent.AuctionFinished;
+import com.example.auction.item.api.ItemEvent.AuctionStarted;
+import com.example.auction.item.api.ItemEvent.ItemUpdated;
+import com.example.auction.item.api.ItemService;
+import com.example.auction.search.api.SearchItem;
+import com.example.auction.search.api.SearchRequest;
+import com.example.auction.search.api.SearchResult;
+import com.example.auction.search.api.SearchService;
+import com.example.elasticsearch.*;
+import com.lightbend.lagom.javadsl.api.ServiceCall;
+import org.pcollections.TreePVector;
+
+import javax.inject.Inject;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+/**
+ *
+ */
+public class SearchServiceImpl implements SearchService {
+
+    public static final String INDEX_NAME = "auction-items";
+    private final Elasticsearch elasticsearch;
+
+    @Inject
+    public SearchServiceImpl(
+            Elasticsearch elasticsearch,
+            ItemService itemService,
+            BiddingService biddingService) {
+        this.elasticsearch = elasticsearch;
+
+        // TODO: use ES' _bulk API
+        itemService.itemEvents().subscribe().atLeastOnce(Flow.<ItemEvent>create().map(this::toDocument).mapAsync(1, this::store));
+        biddingService.bidEvents().subscribe().atLeastOnce(Flow.<BidEvent>create().map(this::toDocument).mapAsync(1, this::store));
```

Comment:
I find it useful to split topic subscribers out into a separate class/Guice component. It makes things easier to test, and it's better separation of responsibilities.